projects/plugins/boost/app/lib/class-transient.php

```diff
@@ -72,6 +74,46 @@ public static function get( $key, $default = null ) {
 		return $data;
 	}
 
+	/**
+	 * Find all cache keys by prefix.
+	 *
+	 * @param string $prefix    Cache key prefix.
+	 *
+	 * return string[] Array of string representing the keys that match the prefix.
+	 */
+	public static function find_keys_by_prefix( $prefix ) {
+		global $wpdb;
+
+		/**
+		 * The prefix used in option_name.
+		 */
+		$option_prefix = static::key( $prefix );
+
+		/*
+		 * Find the length of prefix used by all transients.
+		 */
+		$transient_prefix_length = strlen( static::OPTION_PREFIX );
+
+		/*
+		 * We are looking for all cache entries in the database for supplied cache prefix. As the `Transient` class adds
+		 * another prefix before storing the cache in database, we will remove that during the query to return the
+		 * proper keys.
+		 */
+		$keys = $wpdb->get_col(
+			$wpdb->prepare(
+				"
+					SELECT  SUBSTRING(`option_name`, %d)
```

Comment:
Sorry to do this, but I caught @pyronaur online and during our discussion, he gave a quick drive-by review. He pointed out something I had missed:

This query doesn't have any `LIMIT`, so if there are lots and lots of options which match (there shouldn't be), then it could cause problems. Out of an abundance of caution, could you please add a LIMIT to this (maybe 50?) just to make sure it can't go nuts.

Reply:
I see. It is not going to cover all cases and we will still see some wrong values(in very very rare cases), if there are for example more than 50 URLs used on the site. 
But I assume it's ok on those occasions? Or should we should instead create a delete query?

Reply:
Actually a delete query seems like it would save a lot of back and forth if there are more than 50 entries hanging around. That seems reasonable to me. :)

Reply:
Updated with a delete query. There was also an issue of wildcard match of `_` characters in name that could cause minor issues, which I have addressed. See comment on like [91-92](https://github.com/Automattic/jetpack/pull/20737/files#diff-90b3af2490f0c347b437e55a42ccf178d7eba6d13d7ac5920632a3b275e56000R91-R92).